pkg/recutil/reconcile.go

```diff
@@ -35,13 +38,25 @@ type ObjectReconcileFunc func(logger kitlog.Logger, request reconcile.Request, o
 func ResolveAndReconcile(ctx context.Context, logger kitlog.Logger, mgr manager.Manager, objType runtime.Object, inner ObjectReconcileFunc) reconcile.Reconciler {
 	return reconcile.Func(func(request reconcile.Request) (res reconcile.Result, err error) {
 		logger := kitlog.With(logger, "request", request)
-		logger.Log("event", EventStart, "msg", "starting reconciliation")
+		logger.Log("event", EventRequestStart, "msg", "Reconcile request start")
 
 		defer func() {
 			if err != nil {
-				logger.Log("event", EventError, "error", err)
+				// Conflict errors are typically temporary, caused by something
+				// external to the controller updating the object that is being
+				// reconciled, or the controller's object cache not being up-to-date
+				// (which can occur as a result of an update in a previous
+				// reconciliation).
+				// In these cases the reconciliation will be retried, and we do not
+				// want to pollute the object's events with transient errors which have
+				// no means of avoiding.
+				if apierrors.IsConflict(errors.Cause(err)) {
```

Comment:
This error is fairly well reproducible, in my e2e environment at least. Just run `cat config/samples/workloads_v1alpha1_console.yaml | kubectl apply -f` a few times in succession and you'll hit it.

Reply:
I think this is a very sensible way of handling it.